libsql/internal/http/driver.go

```diff
@@ -66,40 +94,35 @@ func (c *conn) Begin() (driver.Tx, error) {
 	return nil, fmt.Errorf("Begin method not implemented")
 }
 
-func convertArgs(args []driver.NamedValue) params {
-	if len(args) == 0 {
-		return params{}
-	}
-	sortedArgs := [](*driver.NamedValue){}
-	for idx := range args {
-		sortedArgs = append(sortedArgs, &args[idx])
-	}
-	sort.Slice(sortedArgs, func(i, j int) bool {
-		return sortedArgs[i].Ordinal < sortedArgs[j].Ordinal
-	})
-	names := [](string){}
-	values := [](any){}
-	for idx := range sortedArgs {
-		if len(sortedArgs[idx].Name) > 0 {
-			names = append(names, sortedArgs[idx].Name)
-		}
-		values = append(values, sortedArgs[idx].Value)
-	}
-	return params{Names: names, Values: values}
-}
-
 func (c *conn) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Result, error) {
-	_, err := callSqld(ctx, c.url, query, convertArgs(args))
+	rs, err := callSqld(ctx, c.url, query, args)
 	if err != nil {
 		return nil, err
 	}
+
+	if err := assertNoResultWithError(rs, query); err != nil {
+		return nil, err
+	}
+
 	return &result{0, 0}, nil
 }
 
 func (c *conn) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
-	rs, err := callSqld(ctx, c.url, query, convertArgs(args))
+	rs, err := callSqld(ctx, c.url, query, args)
 	if err != nil {
 		return nil, err
 	}
-	return &rows{rs, 0}, nil
+	return &rows{rs, 0, 0}, nil
+}
+
+func assertNoResultWithError(resultSets []httpResults, query string) error {
+	for _, result := range resultSets {
+		if result.Error != nil {
+			return fmt.Errorf("failed to execute SQL: %s\n%s", query, result.Error.Message)
+		}
+		if result.Results == nil {
+			return fmt.Errorf("no results for statement")
+		}
```

Comment:
Is this correct? Are all statements returning results?

Reply:
Yes. Even statements without outputs are returned as a empty result (columns and rows with empty arrays afair)